subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -274,6 +275,10 @@ public void apply(ProjectInternal project) {
         configureTest(project, javaConvention);
         configureArchivesAndComponent(project, javaConvention);
         configureBuild(project);
+
+        if (project.getPluginManager().hasPlugin("java-platform")) {
```

Comment:
:x: Let's use `withPlugin` instead, because this would fail only if you apply `java-platform` first.

Reply:
That's the whole point to not cause a breaking change. I am warning only when it seems to work, that is when `java-platform` is applied first and then `java` is.
The opposite fails with an exception anyway as reflected in the change to `JavaPlatformPlugin`.

Reply:
I think we should always warn. It doesn't make sense to apply both, whatever the condition.